definition_builder.go

```diff
@@ -205,18 +205,27 @@ func (b definitionBuilder) buildProperty(field reflect.StructField, model *spec.
 	}
 
 	// not a primitive
+	// Since the `prop` variable in each of the cases below is a new reference, we need to re-set the `Example` field.
```

Comment:
Changed the previously confusing 2-line comments into a 1-liner. Let me know what you think @emicklei. Thanks!